app/src/main/java/com/example/colecofer/android_audio_visualizer/VisualizerActivity.java

```diff
@@ -151,14 +134,23 @@ protected void onPause() {
         surfaceView.onPause();
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        Log.d("test", "IM BEING DESTROYED");
+        this.mediaPlayer.release();
+    }
+
     @Override
     public void onWaveFormDataCapture(Visualizer visualizer, byte[] waveform, int samplingRate) {
     }
 
     @Override
     public void onFftDataCapture(Visualizer visualizer, byte[] fft, int samplingRate) {
         /** Gives us the decibel level for the fft bucket we care about **/
-        double dbs = utils.getDBs(fft[REAL_BUCKET], fft[IMAGINARY_BUCKET], this.audioSampleSize);
+        double dbs = getDBs(fft[REAL_BUCKET], fft[IMAGINARY_BUCKET], this.audioSampleSize);
+
+        //this.previousUpdateTime = updateDbHistory(dbs, VisualizerModel.getInstance().currentVisualizer.dbHistory, this.previousUpdateTime);
```

Comment:
this should be deleted if its not being used

Reply:
Leaving for the future